File: Go/BuilderPattern/FactoryPattern/FactoryMethod/CoffeeStore.go
package FactoryMethod

type CoffeeStore struct {
	factory CoffeeFactory
}

func (c *CoffeeStore)SetFactory(factory CoffeeFactory) {
	c.factory = factory
}

func (c *CoffeeStore)OrderCoffee() abstractCoffee{
	coffee := c.factory.createCoffee()
	coffee.addMike()
	coffee.addsuger()
	return coffee
}

File: Java/src/main/java/com/design/pattern/builderPattern/FactoryPattern/CoffeDemo/FactoryMethod/CoffeeStore.java
package com.design.pattern.builderPattern.FactoryPattern.CoffeDemo.FactoryMethod;


public class CoffeeStore {
    private CoffeeFactory factory;

    public void setFactory(CoffeeFactory factory) {
        this.factory = factory;
    }

    public Coffee orderCoffee() {
        Coffee c = factory.createCoffee();
        c.addMike();
        c.addSugar();
        return c;
    }
}
